families/supplychain/python/sawtooth_supplychain/rest_api/route_handlers.py

```diff
@@ -0,0 +1,756 @@
+
+    except psycopg2.OperationalError as e:
+        raise e
+
+
+class RouteHandler(object):
+    """Contains a number of aiohttp handlers for endpoints in
+    the Supply Chain Rest Api.
+
+    Each handler takes an aiohttp Request object, and uses the data in
+    that request to create queries to send to the database. The query
+    results are processed, and finally an aiohttp Response object is
+    sent back to the client with JSON formatted data and metadata.
+
+    If something goes wrong, an aiohttp HTTP exception is raised or returned
+    instead.
+
+    Args:
+        stream (:obj: messaging.stream.Stream):
+        timeout (int, optional): The time in seconds before the Api should
+            cancel a request and report that the validator is unavailable.
+    """
+    def __init__(self, loop, db_cnx=None):
+        loop.set_default_executor(ThreadPoolExecutor())
+        self._loop = loop
+        self._db_cnx = db_cnx
+
+    async def list_agents(self, request):
+        """Fetches a paginated list of all Agents. Using the name filter
+        parameter narrows the list to any Agents who have a name which
+        contains a specified case-insensitive string.
+
+        Request:
+            query:
+                - name: a partial name to filter Agents by
+                - count: Number of items to return
+                - min: Id or index to start paging (inclusive)
+                - max: Id or index to end paging (inclusive)
+
+        Response:
+            data: JSON array of Agents
+        """
+
+        # Get query parameters
+        name = request.url.query.get('name', '')
+        count = request.url.query.get('count', DEFAULT_PAGE_SIZE)
+        p_min = request.url.query.get('min', 0)
+        p_max = request.url.query.get('max', None)
+
+        count, p_min, p_max = \
+            RouteHandler._check_paging_params(count, p_min, p_max)
+
+        # Create query
+        name = "%" + name + "%"  # Add pattern matching to search string
+        query_tuple = ()
+
+        if p_max is not None:
+            query = ("SELECT identifier, name FROM agent WHERE name LIKE %s "
+                     "AND id >= %s AND id <= %s ORDER BY id limit %s")
+            query_tuple += (name, p_min, p_max, count)
+        else:
+            query = ("SELECT identifier, name FROM agent WHERE name LIKE %s "
+                     "AND id >= %s ORDER BY id limit %s")
+            query_tuple += (name, p_min, count)
+
+        try:
+            rows = await db_query(self._db_cnx, query, query_tuple)
+
+        except psycopg2.OperationalError:
+            LOGGER.debug("Could not connect to database.")
+            raise errors.DatabaseConnectionError
+
+        fields = ('identifier', 'name')
+        data = self._make_dict(fields, rows)
+
+        # Retrieve the max index possible for paging
+        query = "SELECT COUNT(*) FROM agent;"
+        query_tuple = ()
+        rows = await db_query(self._db_cnx, query, query_tuple)
```

Comment:
I am not a fan of this variable naming. If you glance over lines 127/128, it looks like you are sending the same query twice on 129. I would prefer something like `count_query`.